zerver/tests/test_subs.py

```diff
@@ -2143,19 +2143,24 @@ def test_unarchive_stream_private_and_web_public(self) -> None:
             self.assertNotIn(self.example_user("polonius").id, notified_user_ids)
 
     def test_unarchive_stream(self) -> None:
-        hamlet = self.example_user("hamlet")
+        iago = self.example_user("iago")
         cordelia = self.example_user("cordelia")
 
         stream = self.make_stream("new_stream", is_web_public=True)
         was_invite_only = stream.invite_only
         was_web_public = stream.is_web_public
         was_history_public = stream.history_public_to_subscribers
 
-        self.subscribe(hamlet, stream.name)
+        self.subscribe(iago, stream.name)
         self.subscribe(cordelia, stream.name)
         do_deactivate_stream(stream, acting_user=None)
+
+        self.login_user(iago)
```

Comment:
Rather than using `login_user`, you should use `api_patch` to avoid making the rest of the test stateful as to whether we called this. Generally `login_user` is a bad pattern in long tests, as you have to read the whole test above a given point to see what user is doing a given operation, whereas `api_patch` just includes the user as a parameter and thus is more locally readable.